adam-core/src/main/scala/org/bdgenomics/adam/util/AttributeUtils.scala

```diff
@@ -61,27 +71,37 @@ object AttributeUtils {
    */
   def parseAttribute(encoded: String): Attribute = {
     attrRegex.matches(encoded) match {
-      case Some(m) => createAttribute(m.group(1), m.group(2), m.group(3))
+      case Some(m) => createAttribute(m.group(1), m.group(2), m.group(3), m.group(4))
       case None =>
         throw new IllegalArgumentException(
           "attribute string \"%s\" doesn't match format attrTuple:type:value".format(encoded))
     }
   }
 
-  private def createAttribute(attrTuple: (String, String, String)): Attribute = {
+  private def createAttribute(attrTuple: (String, String, String, String)): Attribute = {
     val tagName = attrTuple._1
     val tagTypeString = attrTuple._2
-    val valueStr = attrTuple._3
+    val tagArrayString = attrTuple._3
+    val valueStr = attrTuple._4
+
+    val fullTagString = if (tagArrayString == null) tagTypeString else
```

Comment:
Instead of checking for a null `String` here, I would probably make `tagArrayString` be `Option[String]` and then change this line to be a fold:

```
val fullTagString = tagArrayString.fold(tagTypeString)(arrayString => "%s:%s".format(tagTypeString, arrayString))
```